src/tests/jobtest.c

```diff
@@ -30,26 +30,131 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strings.h>
 #include <daos.h>
+#include <daos_mgmt.h>
+#include <daos/common.h>
+#include <daos/pool.h>
+
+static char *progname;
+
+void print_usage(void)
+{
+	fprintf(stderr, "Usage: %s -p pool_str [-s nsecs] [-x] "
+			"[-h handles_per_pool]\n", progname);
+}
+
+void cleanup_handles(uuid_t *pool_uuids, int num_pools,
+		     daos_handle_t **pool_handles, int handles_per_pool)
+{
+	int i, rc;
+
+	for (i = 0; i < num_pools; i++) {
+		int idx;
+
+		if (uuid_is_null(pool_uuids[i])) {
+			continue;
+		}
+		for (idx = 0; idx < handles_per_pool; idx++) {
+			if (daos_handle_is_inval(pool_handles[i][idx])) {
+				continue;
+			}
+			rc = daos_pool_disconnect(pool_handles[i][idx], NULL);
+			if (rc) {
+				struct dc_pool	*pool;
+				char		pool_str[64];
+				char		hdl_str[64];
+
+				pool = dc_hdl2pool(pool_handles[i][idx]);
+				uuid_unparse_lower(pool->dp_pool, pool_str);
+				uuid_unparse_lower(pool->dp_pool, hdl_str);
+				printf("disconnect handle %s from pool %s "
+					"failed\n", hdl_str, pool_str);
+			}
+		}
+		free(pool_handles[i]);
+		pool_handles[i] = NULL;
+	}
+}
+
+/*
+ * Takes in a comma separated list of UUIDs and turns them into a list of
+ * UUID structures. It will return the number of entries on success and -1 on
+ * failure
+ */
+int parse_pool_handles(char *hndl_str, uuid_t **handles)
+{
+	char		*idx, *pool_str;
+	const char	delim[2] = ",";
+	int		hdl_count = 1 /* we have at least 1 */, i = 0, rc;
+	uuid_t		*hndls = NULL;
+
+	/* Figure out how many handles are in the string */
+	idx = strchr(hndl_str, ',');
+	while (idx != NULL) {
+		hdl_count++;
+		idx = strchr(idx + 1, ',');
+	}
+
+	hndls = calloc(hdl_count, sizeof(uuid_t));
+	if (!hndls) {
+		perror("Unable to allocate space for pool handle list");
+		return -1;
+	}
+
+	/* Parse the UUIDS into their structures */
+	pool_str = strtok(hndl_str, delim);
+	while (pool_str != NULL) {
+		if (uuid_parse(pool_str, hndls[i]) != 0) {
+			printf("Invalid pool uuid: %s\n", pool_str);
+			rc = -1;
+			goto out_err;
+		}
+		i++;
+		pool_str = strtok(NULL, delim);
+	}
+	rc = hdl_count;
+	*handles = hndls;
+	goto out;
+
+out_err:
+	free(hndls);
+out:
+	return rc;
+}
 
 int
 main(int argc, char **argv)
 {
-	int	rc;
-	int	opt;
-	int	sleep_seconds = 5;
-	int	abnormal_exit = 0;
+	int		rc, opt, i, num_pools;
+	int		sleep_seconds = 5; /* time to sleep before cleanup */
+	int		abnormal_exit = 0; /* whether to kill the process*/
+	int		handles_per_pool = 5; /* Number of pool connections*/
+	int		well_behaved = 0; /* Whether to disconnect at the end*/
```

Comment:
Should be a bool? Looks like abnormal_exit should be as well.

Reply:
While c99 has a bool type on RHEL its defined in C++ headers so I'm not sure we want to use it here. Its also just an int with some enums anyway.

Reply:
Yeah, not that important, but we do use a Linux-defined bool type throughout the rest of DAOS.